servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SourceToFuture.java

```diff
@@ -45,6 +45,9 @@ abstract class SourceToFuture<T> implements Future<T> {
     @Nullable
     private volatile Object value;
 
+    // The timestamp of the last `.get()` call. This is intended to help with debugging stuck threads via heap dumps.
+    private long lastGetTimestamp;
```

Comment:
It's not clear to me if this needs some stronger ordering semantics or not. I imagine there is some sort of store guarantee from `latch.await()` but I'm not _certain_.

Reply:
The current approach lgtm, we should not have cases when multiple threads trying to invoke `get()` at the same future. Most cases call `get()` once and if they need to "retry" (in case of get timeout), they likely gonna do it from the same thread.

Reply:
An alternative implementation might be a bit more desirable: https://github.com/apple/servicetalk/pull/3052. The benefits of that PR are that each call to `future.get()` is tracked but only if we're using the `BlockingUtils`, which should narrow down the usage to blocking client and server operations. It technically an allocate another object (which is likely a drop in the ocean) but that object should be trivial to stack allocate so chances are good it doesn't have any allocation overhead.